### PR TITLE
improve no-trainer examples

### PR DESCRIPTION
**Pull request by pacman100**
# What does this PR do?
1. In no-trainer examples, `train_loss` being logged wasn't normalized and as such wasn't intuitive to understand. This also made it difficult to compare train loss between different tools such as comparing train loss from Trainer with that of Accelerate. This PR normalizes the train_loss per epoch to make is more intuitive and comparable.
2. Replaces HF AdamW with torch AdamW for NLP no-trainer examples. This prevents corresponding warning being displayed.
3. Fixing no-trainer examples so that tracker run is created only for the main process else wandb will create num_processes runs with no data.
4. converting `train_loss` from tensor to float so that it gets logged in tensorboard tracker
5. Fixing `run_ner_no_trainer.py` to correctly log `train_loss` in `all_results.json`
6. Adding `report_to` arg to enable users to specify preferred tracker instead of all available trackers which is default option. This prevents logging to the trackers that user doesn't want.
7. In many no-trainer NLP tasks one can train model from scratch, this means that user can bypass `model_name_or_path` arg. However, it is set as required for all scripts which throws error when it isn't specified. Setting this arg `required=False` in corresponding examples to resolve the error when training from scratch.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for all the improvements!

File: examples/pytorch/image-classification/run_image_classification_no_trainer.py [hunk 1]
```diff
@@ -386,10 +399,11 @@ def collate_fn(examples):
...
     if args.with_tracking:
-        experiment_config = vars(args)
-        # TensorBoard cannot log Enums, need the raw value
-        experiment_config["lr_scheduler_type"] = experiment_config["lr_scheduler_type"].value
-        accelerator.init_trackers("image_classification_no_trainer", experiment_config)
+        if accelerator.is_main_process:
+            experiment_config = vars(args)
+            # TensorBoard cannot log Enums, need the raw value
+            experiment_config["lr_scheduler_type"] = experiment_config["lr_scheduler_type"].value
+            accelerator.init_trackers("image_classification_no_trainer", experiment_config)
```
Comment (muellerzr) [inline examples/pytorch/image-classification/run_image_classification_no_trainer.py:406]: Would it make more sense to have init_trackers only work on process 0 in this case? Rather than having the user do this

Comment (sgugger) [reply examples/pytorch/image-classification/run_image_classification_no_trainer.py:406]: Oh definitely!

Comment (pacman100) [reply examples/pytorch/image-classification/run_image_classification_no_trainer.py:406]: Yes, that would be better way of handling this because `accelerator.log` only logs only main process.

However, there are scenarios wherein one would want to log to multiple processes. For example, while experimenting with FSDP, I wanted to log GPU memory consumed by each GPU process. The logger would write the GPU memory stats to separate text files with one for each process.

Comment (muellerzr) [reply examples/pytorch/image-classification/run_image_classification_no_trainer.py:406]: Let's include a small comment for why we want to do that in this case then, and think about expanding the tracking docs to discuss this

Comment (pacman100) [reply examples/pytorch/image-classification/run_image_classification_no_trainer.py:408]: Added comment.

**Comment by pacman100** [comment]
Opened a new PR on the branch of main repo instead of fork #17449
